### PR TITLE
Native copy and recursive directories creation

### PR DESCRIPTION
**Pull request by paillechat**

### CODE REVIEW

**Comment by frankdejonge** [comment]
@paillechat Hi, wouldn't this always be the better way to copy a file? Or are there webdav implementations that don't support this?

**Comment by paillechat** [comment]
@frankdejonge Hi! Native webdav copy is faster because we don't need to download target file and upload it by the new way. We just copy file on server without download. But default for the backward compatibility library is working as it was working before via download&upload.

**Comment by frankdejonge** [comment]
@paillechat the benefit is quite obvious so I'm all for just having this patch in, I just wondered why we'd even keep the old behavior.

**Comment by paillechat** [comment]
@frankdejonge cool :)

File: src/WebDAVAdapter.php [hunk 1]
```diff
@@ -126,6 +135,10 @@ public function read($path)
      */
     public function write($path, $contents, Config $config)
     {
+        if (!$this->createDir(dirname($path), $config)) {
```
Comment (frankdejonge) [inline src/WebDAVAdapter.php]: Dirname returns `.` when there's no parent directory, the `Util::dirname` normalizes this.

Comment (paillechat) [inline src/WebDAVAdapter.php]: ok, fixed

**Comment by harpax** [comment]
is this one ready to be merged? Looking forward to the recursive path creation..

**Comment by S3RK** [comment]
@frankdejonge Hi. Also waiting for this PR. Can you merge it?

**Comment by omnilight** [comment]
Hi! We are also waiting for this to be merged.

**Comment by frankdejonge** [comment]
Tagged as 1.0.5

**Pull request merged**